### PR TITLE
ENH: Added OldVonHamos4Crystal and OldVonHamosCrystal classes to pcds…

### PR DESCRIPTION
**Pull request by jortiz-slac**
…devices/spectrometer.py

## Description
Added OldVonHamos4Crystal and OldVonHamosCrystal classes to pcdsdevices/spectrometer.py

## Motivation and Context
This will allow scientists/users to manipulate the old four crystal Von Hamos interactively in a hutch python session.

## How Has This Been Tested?
Classes were tested using simulated motors then tested on actual Von Hamos hardware using a ipython session.
All axes were tested.

## Where Has This Been Documented?
Doctrings used in classes.
modification and tests are documented in JIRA https://jira.slac.stanford.edu/browse/LCLSECSD-1513

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [NA] Test suite passes locally
- [NA ] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by tangkong — COMMENTED** [review]
Looks good on the whole to me, thanks for documenting everything so nicely. 

From what I can see here, it looks like the main change is in how the PV's are structured.  The python-side interface of this `OldVonHamosCrystal` is similar to the existing `VonHamos4Crystal`

Just a few clarifying comments from me here.  If it's been tested and works against the real device I don't have too many other concerns

File: pcdsdevices/device_types.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
 from .sensors import RTD, TwinCATThermocouple
 from .sequencer import EventSequencer
 from .slits import Slits
-from .spectrometer import Kmono, VonHamos4Crystal
+from .spectrometer import Kmono, VonHamos4Crystal, OldVonHamos4Crystal
```
Comment (tangkong) [inline pcdsdevices/device_types.py:45]:
```suggestion
from .spectrometer import Kmono, OldVonHamos4Crystal, VonHamos4Crystal
```

The pre-commit checks failed here, asking for these to be listed in alphabetical order

**Comment by silkenelson** [comment]
Regarding the name: I veto'ed the original name ( as I did not think it was meaningful ), but did not have a much better idea, so I could use help:
This is a 4-crystal von Hamos that uses Kohzu stages. It uses the very first Beckhoff motion PLC w/ IOC that Zach built (pre ADS) etc, so it is for an 'old' Beckhoff motion setup.

File: pcdsdevices/spectrometer.py [hunk 2]
```diff
@@ -361,3 +362,45 @@ class HXRSpectrometer(BaseInterface, GroupDevice):
     removed = False
     transmission = 1
     SUB_STATE = 'state'
+
+
+class OldVonHamosCrystal(BaseInterface, GroupDevice):
```
[the remark this answers is not in the document]
Comment (jortiz-slac) [reply pcdsdevices/spectrometer.py:367]: I spoke with Vincent and Silke about the name and agreed 'old' was okay - definitely better than using 'Mfx' 
This is an old PLC system that will not be re-used, but this legacy Von Hamos will be used for some time.

**Comment by tangkong** [comment]
Is the existing vonHamos class the one being used for MFX?  Or will there be a third class created for that system?

Putting a "generation 1 PLC" note in the docstring would probably satisfy me.   Maybe we name the class `Gen1VonHamosCrystal` or something similar?

**Comment by jortiz-slac** [comment]
I do like Gen1, If everyone agrees, we can change it to that

**Comment by silkenelson** [comment]
I second Gen1, looks clearly more professional than 'old'.

**Comment by vespos** [comment]
Did we try the generation of the auto-generated typhos screen from this class? That may just be my ignorance here, but I am not familiar with how the nested devices would show up on the screen. Also, shouldn't the kind be set to "hinted" for the component to appear in the screen?

Else this all looks good to me! The screen is definitely a nice to have, but not critical as this is mostly used at XCS, where the CLI is generally used.

File: pcdsdevices/spectrometer.py [hunk 3]
```diff
@@ -361,3 +362,45 @@ class HXRSpectrometer(BaseInterface, GroupDevice):
...
+class Gen1VonHamosCrystal(BaseInterface, GroupDevice):
+    """Pitch, yaw, and translation motors for control of a single crystal."""
+
+    tab_component_names = True
+
+    pitch = FCpt(EpicsMotorInterface, '{prefix}{_pitchAxis}', kind='normal')
+    yaw = FCpt(EpicsMotorInterface, '{prefix}{_yawAxis}'  , kind='normal')
+    trans = FCpt(EpicsMotorInterface, '{prefix}{_transAxis}', kind='normal')
+
+    def __init__(self, prefix, pitchAxis, yawAxis, transAxis, **kwargs):
```
Comment (ZLLentz) [inline pcdsdevices/spectrometer.py:376]: Small quibble here, these variable names violate the pep8 conventions
https://peps.python.org/pep-0008/#function-and-variable-names

**Comment by ZLLentz** [comment]
> Did we try the generation of the auto-generated typhos screen from this class? That may just be my ignorance here, but I am not familiar with how the nested devices would show up on the screen. Also, shouldn't the kind be set to "hinted" for the component to appear in the screen?

Here's a few screenshots of what typhos shows for `typhos --fake-device pcdsdevices.spectrometer.Gen1VonHamos4Crystal`
It essentially puts the common motor first, then makes a subheading for each crystal that each contain the motors for that crystal. I think we want the auto-generation to do these sorts of devices a bit differently but that's how it runs for now.
I guess you don't need to put subdevices as hinted.

image
image

File: pcdsdevices/spectrometer.py (hunk 3 above)
Comment (jortiz-slac) [reply pcdsdevices/spectrometer.py]: Hi Zach, no problem. Changed them to be PEP 8 – Style Guide for Python Code compliant.
Thanks!

**Review by ZLLentz — APPROVED** [review]
Looks good to me, I see no reason not to merge

**Pull request merged**